sys/posix/sockets/posix_sockets.c

```diff
@@ -940,6 +941,8 @@ static ssize_t socket_recvfrom(socket_t *s, void *restrict buffer,
         }
     }
 
+    bool peek_mode = (flags & MSG_PEEK) != 0;
```

Comment:
```suggestion
    bool peek_mode = (flags & MSG_PEEK) && IS_USED(MODULE_SOCK_AUX_PEEK);
```

Reply:
Also, with lazy evaluation, wouldn't it be more effective the other way around?

Reply:
If it's compile-time const it will be optimized away either way.

Reply:
> Also, with lazy evaluation, wouldn't it be more effective the other way around?

Does that really make a difference? Isn't that optimized in the end anyway?

Reply:
Aside from that, while peek is optional for the `lwip_sock_*`, it is mandatory for `posix_socket` (see `USEMODULE += sock_aux_peek` in `sys/Makefile.dep`). I don't think it makes sense to split posix features since users expect them to work when the flag is available.